cep-9999.md

```diff
@@ -0,0 +1,55 @@
+# CEP ???? - Build provenance metadata
+
+<table>
+<tr><td> Title </td><td> CEP ???? -  Build provenance metadata </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Jaime Rodríguez-Guerra &lt;jaime.rogue@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> Mar 10, 2025</td></tr>
+<tr><td> Updated </td><td> Mar 10, 2025</td></tr>
+<tr><td> Discussion </td><td> N/A </td></tr>
+<tr><td> Implementation </td><td> https://github.com/conda-forge/conda-smithy/pull/1577 </td></tr>
+</table>
+
+## Abstract
+
+This CEP aims to standardize how the conda ecosystem leverages free-form recipe metadata
+to annotate provenance of its published artifacts. Two features are applied:
+
+- `extra` section in the recipe metadata
+- `extra` metadata in `about.json` (via CLI flag `--extra-meta`)
+
+## Motivation
+
+Provenance metadata is useful to assess how and when a conda artifact was built.
+
+Since late 2023, thanks to conda-smithy 3.28.0 and later, conda-forge feedstocks have been
+adding CI provenance in the produced artifacts. `defaults` also applies the same conventions.
+This is used by apps like `conda-metadata-app` to show provenance information in the build details. See the table in this [Python 3.13 example](https://conda-metadata-app.streamlit.app/?q=conda-forge%2Flinux-64%2Fpython-3.13.2-hf636f53_101_cp313.conda).
+
+Additional provenance metadata can be collected for source origins, which can
+be useful for efforts like dependency mapping across ecosystems. See
+[PEP 725](https://peps.python.org/pep-0725/) for practical applications in the
+context of PyPI <> conda interoperability. This type of provenance is out of
+scope for this CEP and may be discussed separately.
+
+## Specification
+
+CI pipelines MUST instrument the build workflows to add the following metadata keys
+via `--extra-meta` or equivalent:
+
+- `sha`: Required. Commit hash of the feedstock being built.
+- `remote_url`: Required on CI. Repository URL of the feedstock being built.
+- `flow_run_id`: Optional. CI-specific identifier for the workflow run.
```

Comment:
The name `flow_run_id` originates from Anaconda's current build system. I wish that we could use something more generic and meaningful. But I guess the boat has sailed already? Or do you foresee a future where CF would be willing to change this to something else (on our side at Anaconda, this is very easy to do).

Reply:
We can change it easily too, but then we will have to maintain two ways of accessing this metadata because the already stamped artifacts won't be rebuilt. I think `flow_run_id` is sufficiently generic. I always read it as "(work)flow run ID".